### PR TITLE
[ML] Feature importance unit test for missing data

### PR DESCRIPTION
**Pull request by valeriy42**
This PR tests that feature importance behaves correctly in the face of missing data values.

### CODE REVIEW

File: lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc [hunk 1]
```diff
@@ -65,6 +65,36 @@ void setupLinearRegressionData(const TStrVec& fieldNames,
...
+        return core::CStringUtils::typeToStringPrecise(result, core::CIEEE754::E_DoublePrecision);
+    };
+
+    for (std::size_t i = 0; i < rows; ++i) {
+        TDoubleVec regressors;
+        rng.generateUniformSamples(0.0, 10.0, cols - 1, regressors);
+
+        fieldValues[0] = target(regressors);
+        for (std::size_t j = 0; j < regressors.size(); ++j) {
+            double regressor = regressors[j] > 9.0 ? core::CDataFrame::valueOfMissing()
```
Comment (tveasey) [inline lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc]: I think if you want to pass missing as a string you need to pass an empty string, i.e. something like:
```
if (regressors[j] <= 9.0) {
    fieldValues[j + 1] = core::CStringUtils::typeToStringPrecise(
                             regressor, core::CIEEE754::E_DoublePrecision);
}
```
(p.s. The approach you use now may work, but it's non-standard.)

File: lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc [hunk 2]
```diff
@@ -325,4 +379,38 @@ BOOST_FIXTURE_TEST_CASE(testRunBoostedTreeRegressionFeatureImportanceNoShap, SFi
     }
 }
 
+BOOST_FIXTURE_TEST_CASE(testMissingFeatures, SFixture) {
+    // Test that feature importance behaves correctly when some features are missing:
+    // We randomly 10% of all data in a simple additive model target=c1+c2+c3+c4. Hence,
```
Comment (tveasey) [inline lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc]:
```suggestion
    // We randomly omit 10% of all data in a simple additive model target=c1+c2+c3+c4. Hence,
```

File: lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc (hunk 1 above)
Comment (valeriy42) [reply lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc]: Fixed in 4eb14fdd

**Comment by valeriy42** [comment]
Thank you @tveasey for the review. I adjusted the code as you suggested.

**Pull request merged**